packaging/linux/post_install.sh

```diff
@@ -1,56 +1,79 @@
 # and rpm-specific post-install scripts call into this after doing their
 # distro-specific work, which is mainly setting up package repos for updates.
 
 set -u
 
-# Create the /keybase symlink to the first mountpoint, if it doesn't
-# already exist. We can't use a regular [ -e ... ] check, because
-# stat'ing /keybase fails with a permissions error when kbfsfuse is
-# mounted, so that check always returns false. Instead we check
-# whether mkdir succeeds.
-rootlink="/keybase"
-vardir="/var/lib/keybase"
-mount1="$vardir/mount1"
-sample="/opt/keybase/mount-readme"
-khuser="keybasehelper"
-khbin="/usr/bin/keybase-mount-helper"
-
-# Create the keybasehelper system user, without login privileges.
-if useradd --system -s /bin/false -U -M $khuser &> /dev/null ; then
-    echo Created $khuser system user for managing mountpoints.
-fi
+rootmount="/keybase"
+krbin="/usr/bin/keybase-redirector"
 
-chown "$khuser":"$khuser" "$khbin"
-chmod 4755 "$khbin"
-if mkdir $vardir &> /dev/null ; then
-    ln -s "$sample" "$mount1"
-    chown -R "$khuser":"$khuser" "$vardir"
-fi
+vardirDeprecated="/var/lib/keybase"
+khuserDeprecated="keybasehelper"
+khbinDeprecated="/usr/bin/keybase-mount-helper"
+optDeprecated="/opt/keybase/mount-readme"
+
+chown root:root "$krbin"
+chmod 4755 "$krbin"
 
-currlink=`readlink $rootlink`
-if [ -z "$currlink" ] ; then
-    if fusermount -uz "$rootlink" &> /dev/null ; then
+run_redirector() {
+  logdir="${XDG_CACHE_HOME:-$HOME/.cache}/keybase"
+  mkdir -p "$logdir"
+  nohup "$krbin" "$rootmount" >> $logdir/keybase.redirector.log 2>&1 &
+}
+
+currlink=`readlink -m "$rootmount"`
+if [ -n "$currlink" ] ; then
+    # Upgrade from a rootlink-based build.
+    if rm "$rootmount" &> /dev/null ; then
+        echo Replacing old $rootmount symlink.
+    fi
+    mounts=`cat /proc/mounts | awk '{print $2}'`
+    # TODO: Set $IFS in case any of the mountpoints contain a space?
+    # Probably not a big deal since all this loop does is start the
+    # redirector, which would also be done by the next call to
+    # `run_keybase`.
+    for m in $mounts; do
+      if [ "$m" = "$currlink" ]; then
+        echo Starting root redirector at $rootmount.
+        run_redirector
+        break
+      fi
+    done
+elif [ -d "$rootmount" ] ; then
+    # Handle upgrading from old builds that don't have the rootlink.
+    currowner=`stat -c %U "$rootmount"`
+    if [ $currowner != "root" ]; then
         # Remove any existing legacy mount.
-        echo Unmounting $rootlink...
+        echo Unmounting $rootmount...
         if killall kbfsfuse &> /dev/null ; then
             echo Shutting down kbfsfuse...
         fi
+        rmdir "$rootmount"
         echo You must run run_keybase to restore file system access.
+    elif killall `basename "$krbin"` &> /dev/null ; then
+        # Restart the root redirector in case the binary has been updated.
+        fusermount -u "$rootmount"
+        run_redirector
```

Comment:
If there were a systemd service for the redirector, I think this could be a systemd restart.